content-metadata-protobuf/proto/Video.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto2";
+
+// Publication status before joystream
+message PublishedBeforeJoystream {
+    // Was video published before joystream platform
+    optional bool is_published = 1;
```

Comment:
is this really needed? coudlnt the whole `PublishedBeforeJoystream` actually just be an optional timestamp? that covers every state.

Reply:
Not quite. So it being an `optional`  field in the protobuf message is sufficient to inform the consumer that we want to change or set the publishedbeforejoystream attribute of the metadata, but the the value is itself is an optional to have assocaited with a video.  ie. we need `Option<Option<timestamp>>` in rust syntax.

I could have chosen a simpler representation of just `optional u32 timestamp` and application would have to interpret the value 0 for example to mean that a video was not published before joystream. But that doesn't work if a video came before the unix timestamp start date..